kubemarine/procedures/add_node.py

```diff
@@ -101,10 +101,13 @@ def cache_installed_packages(cluster):
 
     # avoid caching when unchanged nodes are not equal to GLOBAL os
     global_os = system.get_os_family(cluster)
-    for node in cluster.nodes['all'].get_unchanged_nodes().get_ordered_members_list(provide_node_configs=True):
-        if cluster.context["nodes"][node['connect_to']]["os"]['family'] != global_os:
+    for nodes in cluster.nodes['all'].get_first_member().get_ordered_members_list(provide_node_configs=True):
+        version_os = cluster.context["nodes"][nodes['connect_to']]["os"]['version']
+    for node in cluster.nodes['all'].get_changed_nodes().get_ordered_members_list(provide_node_configs=True):
+        if cluster.context["nodes"][node['connect_to']]["os"]['family'] != global_os \
+                or cluster.context["nodes"][node['connect_to']]["os"]['version'] != version_os:
```

Comment:
The idea is good, but

1. It looks strange to take global_os and version_os from different sources but use the similar way.
2. It does not make sense to compare changed_nodes (new_nodes actually) and global_os, as global_os is calculated based on new nodes only. So the != comparison result is always false.
3. The following case is not covered: initial nodes have OS2 and OS1. New node has OS2. version_os fill be detected by first node which is OS2. The != comparison with changed (new) nodes will be false. Packages are cached but isn't it  a problem that there are finally cluster with different OS: OS2, OS1, OS2?

Reply:
Fix code